docs/content/how-mixins-work.md

```diff
@@ -0,0 +1,119 @@
+---
+
+## What is a Mixin
+
+A cloud native application bundle built with Porter consists of two main components: the Porter runtime and a declarative manifest file named `porter.yaml`.
+
+The Porter runtime provides the entry point for the bundle and is responsible for executing the desired functionality that has been expressed in the `porter.yaml`. The `porter.yaml` declares what should happen for bundle action. Each bundle action is defined as one or more operations, or steps. Each step definition defines a discrete piece of functionality, such installing a Helm chart or creating a service in a cloud provider. The Porter runtime is responsible for executing each step, but does not implement the desired functionality itself. The functionality declared in each step is actually provided by components called mixins. Porter invokes the mixin by passing the relevant section of the `porter.yaml` to the mixin via standard input. The mixin accepts the YAML document that describes the desired result and performs the desired action. Once finished, the mixin wil write any desired outputs to standard out and return control to the Porter runtime.
+
+For example, a bundle author may wish to execute a bash command. The author would include the `exec` mixin in the mixins section of the porter.yaml and then create a step that defines the desired bash command. For example, the following bundle example would execute `bash -c "echo Hello World"`:
+
+```
+mixins:
+- exec
+
+name: hello
+version: 0.1.0
+invocationImage: porter-hello:latest
+
+install:
+- description: "Say Hello"
+  exec:
+    command: bash
+    arguments:
+      - -c
+      - echo Hello World
+```
+
+## Mixin API
+
+Porter provides a contract that mixins must fulfill in order to be included in the Porter ecosystem. This contract specifies how Porter will execute the mixins, as show above, but also specifies how the mixin is used to build the invocation image for the bundle. Additionally, the contract specifies how a mixin can specify the inputs it can accept and the outputs that it can provide.
+
+### Build Time
+
+The previous example introduced how mixins are used when a bundle is executed. In the case of the `exec` mixin, the resulting bundle invocation image already has everything needed to execute the mixin. Other mixins may require additional runtime software. The `helm` mixin, for example, requires the [Helm](https://helm.sh/) client at runtime. Porter is responsible for building the invocation image for the bundle, so it needs to know what each mixin will need so that it can be included in the bundle. The mixin is responsible for providing any relevant lines to ensure that the generated invocation image Dockerfile has all required runtime components.
+
+In order to provide these lines, Porter expects that a mixin will provide a `build` command. This command should output any necessary Dockerfile commands to standard out. To see this in action, consider the `helm` mixin:
```

Comment:
"in order to provide these lines" this is a bit awkward wording and I'm not sure what it means? The other part of the sentence makes sense without it.